src/views/Pools/hooks/useApprove.tsx

```diff
@@ -95,23 +95,25 @@ export const useVaultApprove = (vaultKey: VaultKey, setLastUpdated: () => void)
 }
 
 export const useCheckVaultApprovalStatus = (vaultKey: VaultKey) => {
-  const [isVaultApproved, setIsVaultApproved] = useState(false)
   const { account } = useWeb3React()
   const cakeContract = useCake()
   const vaultPoolContract = useVaultPoolContract(vaultKey)
-  const { lastUpdated, setLastUpdated } = useLastUpdated()
-  useEffect(() => {
-    const checkApprovalStatus = async () => {
-      try {
-        const currentAllowance = await cakeContract.allowance(account, vaultPoolContract.address)
-        setIsVaultApproved(currentAllowance.gt(0))
-      } catch (error) {
-        setIsVaultApproved(false)
-      }
-    }
 
-    checkApprovalStatus()
-  }, [account, cakeContract, vaultPoolContract, lastUpdated])
+  const key = useMemo<UseSWRContractKey>(
+    () =>
+      account
+        ? {
+            contract: cakeContract,
+            methodName: 'allowance',
+            params: [account, vaultPoolContract.address],
+          }
+        : null,
+    [account, cakeContract, vaultPoolContract.address],
+  )
+
+  const mutate = useMutateSWRContract(key)
+
+  const { data } = useSWRContract(key)
```

Comment:
what's the purpose of  `useMutateSWRContract`?
We can get mutate from useSWRContract

```suggestion
  const { data, mutate } = useSWRContract(key)
```

Reply:
Don't know it already has the `mutate`. Will remove it.